ci-operator/step-registry/cucushift/installer/rehearse/azure/ipi/ovn/hypershift/etcd-encryption/health-check/cucushift-installer-rehearse-azure-ipi-ovn-hypershift-etcd-encryption-health-check-ref.yaml

```diff
@@ -0,0 +1,18 @@
+ref:
+  as: cucushift-installer-rehearse-azure-ipi-ovn-hypershift-etcd-encryption-health-check
+  from_image:
+    namespace: ocp
+    name: "4.16"
+    tag: upi-installer
+  timeout: 15m
+  grace_period: 5m
+  commands: cucushift-installer-rehearse-azure-ipi-ovn-hypershift-etcd-encryption-health-check-commands.sh
+  resources:
+    requests:
+      cpu: 10m
+      memory: 100Mi
+  documentation: |-
+    This step performs health checks on a Hypershift hosted cluster on Azure with ETCD encryption.
+    Assumptions: 
+    1. $KUBECONFIG points to the management cluster.
+    2. ${SHARED_DIR}/nested_kubeconfig contains the kubeconfig of the hosted cluster.
```

Comment:
I am deeply concerned about this. If we need to add a new health check every time we add a new workflow, will this lead to a large number of health checks in the future? The correct approach should be to merge them into cucushift-hypershift-extended-health-check, right? @jianlinliu Is my understanding correct?

Reply:
From my experience or from installer team experience, we are develop all kinds of check steps for different configurations, different install workflow are supposed to reference different check point step per its configurations. E.g: ci-operator/step-registry/cucushift/installer/check/, we do not want to bother health check for a specific configuration check point, generally a health-check step should be shared across many different install workflow just as a common part.

Reply:
@LiangquanLi930 

1. Like I said in our discussion yesterday, it's not only unnecessary but also unmaintainable to put all sorts of health checks together into the same step that runs everywhere. The one I'm adding is for ETCD encryption + Azure ONLY. If we were to put ALL health checks into the same place, then complex conditionals would be required (which part runs on which clusters), and the resulting monolithic step would be unnecessarily complex as Hypershift evolves rapidly and supports more configurations. 

2. As much as I respect @jianlinliu's work and am grateful for his detailed review, I believe this should be a decision for the Hypershift team.

Reply:
From the step naming and location, I suggest to rename the step name to reflect the check point, "health-check" is too common.

And for the path where are supposed to save install workflow and detailed steps (e.g: `create-keys` and `health-check` steps), I still strongly recommend to follow the same convention like what we did before as a whole team, not save the detailed steps into `installer/relearse` folder.

Reply:
Yes, I strongly agree with Jianlin's view. HyperShift is another installer, and as part of the installer team, we should adhere to a unified style and standards.

Reply:
@LiangquanLi930 

1. I discussed this with Jianlin today, and he's fine with the current approach.
2. The similarity between both teams doesn't mean we have to take the exact same approach in every aspect. There's no one-size-fits-all answer.